lib/CMakeLists.txt

```diff
@@ -112,211 +101,10 @@ if (SWIFT_SWIFT_PARSER)
-
-  target_compile_options(${name} PRIVATE $<$<COMPILE_LANGUAGE:Swift>:-target;${target}>)
-
-  # Workaround to touch the library and its objects so that we don't
-  # continually rebuild (again, see corresponding change in swift-syntax).
-  add_custom_command(
-      TARGET ${name}
-      POST_BUILD
-      COMMAND "${CMAKE_COMMAND}" -E touch_nocreate $<TARGET_FILE:${name}> $<TARGET_OBJECTS:${name}>
-      COMMAND_EXPAND_LISTS
-      COMMENT "Update mtime of library outputs workaround")
-
-  # Link against swift-syntax libraries.
-  # TODO: Change to target_link_libraries when cmake is fixed
-  force_target_link_libraries(${name} PUBLIC
-    ${SWIFT_SYNTAX_TARGETS}
-  )
-
-  # Add dependency on the copied swift-syntax libraries.
-  add_dependencies(${name} swiftSyntaxLibraries)
-
-  # Link against dependencies.
-  target_link_libraries(${name} PUBLIC
-    ${APSHL_DEPENDENCIES}
-  )
-  # TODO: Change to target_link_libraries when cmake is fixed
-  force_target_link_libraries(${name} PUBLIC
-    ${APSHL_SWIFT_DEPENDENCIES}
-  )
-
-  # Make sure we can use the host libraries.
-  target_include_directories(${name} PUBLIC
-    ${SWIFT_HOST_LIBRARIES_DEST_DIR})
-
-  if(APSHL_EMIT_MODULE)
-    # Determine where Swift modules will be built and installed.
-
-    set(module_triple ${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_ARCH_${SWIFT_HOST_VARIANT_ARCH}_MODULE})
-    set(module_dir ${SWIFT_HOST_LIBRARIES_DEST_DIR})
-    set(module_base "${module_dir}/${name}.swiftmodule")
-    set(module_file "${module_base}/${module_triple}.swiftmodule")
-    set(module_interface_file "${module_base}/${module_triple}.swiftinterface")
-    set(module_sourceinfo_file "${module_base}/${module_triple}.swiftsourceinfo")
-
-    set_target_properties(${name} PROPERTIES
-        # Set the default module name to the target name.
-        Swift_MODULE_NAME ${name}
-        # Install the Swift module into the appropriate location.
-        Swift_MODULE_DIRECTORY ${module_dir}
-        # NOTE: workaround for CMake not setting up include flags.
-        INTERFACE_INCLUDE_DIRECTORIES ${module_dir})
-
-    # Create the module directory.
-    add_custom_command(
-        TARGET ${name}
-        PRE_BUILD
-        COMMAND "${CMAKE_COMMAND}" -E make_directory ${module_base}
-        COMMENT "Generating module directory for ${name}")
-
-    # Configure the emission of the Swift module files.
-    target_compile_options("${name}" PRIVATE
-        $<$<COMPILE_LANGUAGE:Swift>:
-        -module-name;$<TARGET_PROPERTY:${name},Swift_MODULE_NAME>;
-        -enable-library-evolution;
-        -emit-module-path;${module_file};
-        -emit-module-source-info-path;${module_sourceinfo_file};
-        -emit-module-interface-path;${module_interface_file}
-        >)
-  endif()
-
-  # Export this target.
-  set_property(GLOBAL APPEND PROPERTY SWIFT_EXPORTS ${name})
-endfunction()
+  # Add copied SwiftSyntax libraries to global dependencies.
+  list(APPEND LLVM_COMMON_DEPENDS swiftSyntaxLibraries)
```

Comment:
`add_pure_swift_host_library` targets don't depends on `LLVM_COMMON_DEPENDS`. I will add it.